docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -381,23 +408,46 @@ spec:
 
 * Infrastructure templates are expected to be immutable, so infrastructure template contents do not have to hashed in order to detect
   changes.
-   
+
+##### KubeadmControlPlane rollout (using delete-and-create)
+
+KCP rollout delete-and-create is recommended for resource constrained environment like bare-metal, OpenStack or vSphere resource pools, etc.
+
+- Triggered by:
+    - Changes to Version
+    - Changes to the kubeadmConfigSpec
+    - Changes to the infrastructureRef
+    - `KubeadmControlPlane.Spec.UpgradeStrategy` is set
+
+- Rollout operations rely on scale up and scale down which are be blocked based on Etcd and control plane health checks.
+
+- The algorithm is the following:
+  - Verify control plane replica count is >= 3
+  - Scale down control plane by removing one of the machine that needs rollout
+  - Scale up control plane creating a machine with the new spec
+  - If `KubeadmControlPlane.Spec.UpgradeStrategy` is set and replica count is 1
+    - Prevent control plane to scale down and always force to scale up.
+
+###### Constraints and Assumptions
+
+* If infrastructure does not have resource to scale up when replica count is 1 and scale up is forced, rollout will wait free resource to be able to scale up.
```

Comment:
Uhhm, I'm not sure I understand this constraint, given that scale is allowed only with replicas >= 3

Reply:
Based on my comment above, I would keep this as it is.